File: src/org/likexin/binarytree/MaxDepth.java
package org.likexin.binarytree;

/**
 * http://www.lintcode.com/en/problem/maximum-depth-of-binary-tree/
 *
 * @author Shingo Lee
 */
public class MaxDepth {

    /**
     * 左右递归，注意最后应该把根结点加上。
     *
     * @param root 二叉树根结点
     * @return 二叉树的深度
     */
    public int maxDepth(TreeNode root) {
        if (root == null) {
            return 0;
        }
        int left = maxDepth(root.left);
        int right = maxDepth(root.right);
        return Math.max(left, right) + 1;
    }

}

File: Python/DFS/MaxDepth.py
"""
Definition of TreeNode:
class TreeNode:
    def __init__(self, val):
        self.val = val
        self.left, self.right = None, None
"""

class Solution:
    """
    @param root: The root of binary tree.
    @return: An integer
    """
    def maxDepth(self, root):
        # write your code here
        depth = 1
        if not root:
          return 0
        depth += max(self.maxDepth(root.left), self.maxDepth(root.right))
        return depth
        

